### ISSUE TITLE
why in my model_search  only loss.backward(retain_graph=True) can run

### ISSUE THREAD

**Issue by xiaoiker** [quark0/darts#56]
RT, I am trying to train it bymyself. But I change it for pytorch0.4. But in my model_search  only loss.backward(retain_graph=True) can run. If it is loss.backward( ), is would raise an error that: the second time the graph has been freed. Can someone tell me why?

**Comment by chrischute** [comment]
If you're searching for an RNN cell, you are likely backpropagating all the way back through time ([relevant comment](https://github.com/quark0/darts/blob/master/rnn/train_search.py#L201-L202)). Try changing `repackage_hidden` in `rnn/utils.py` to this:

```python
def repackage_hidden(h):
  if isinstance(h, torch.Tensor):
     return h.detach()
  else:
    return tuple(repackage_hidden(v) for v in h)
```
